api/metric/updowncounter.go

```diff
@@ -0,0 +1,97 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metric
+
+import (
+	"context"
+
+	"go.opentelemetry.io/otel/api/kv"
+)
+
+// Float64UpDownCounter is a metric instrument that sums positive and
+// negative floating point values.
+type Float64UpDownCounter struct {
+	syncInstrument
+}
+
+// Int64UpDownCounter is a metric instrument that sums non-negative
+// integer values.
```

Comment:
I don't think this is restricted to non-negative values only, right? The `UpDownCounter` is not guaranteed to be monotonic I thought.

```suggestion
// Int64UpDownCounter is a metric instrument that sums integer
// values.
```

Reply:
Done.